src/Equinox.Cosmos/Cosmos.fs

```diff
@@ -1157,32 +1195,39 @@ type Connector
 //            co.TransportClientHandlerFactory <- inhibitCertCheck
         co
 
-    /// Yields a CosmosClient configured and connected the requested `discovery` strategy
-    member __.CreateClient
+    abstract member CreateClient: name: string * discovery: Discovery * dbName: string * containerName: string * ?provisioningMode: Provisioning * ?createStoredProcedure: bool * ?storedProcedureName: string * ?skipLog: bool -> EquinoxCosmosClient
+    default __.CreateClient
         (   /// Name should be sufficient to uniquely identify this connection within a single app instance's logs
             name, discovery : Discovery,
+            dbName: string,
+            containerName: string,
+            /// If provided, the database and container will be initialized based on the provided values
+            ?provisioningMode: Provisioning,
+            /// <c>true</c> to create the sync stored procedure during initialization
+            ?createStoredProcedure: bool,
+            /// If provided along with <c>createStoredProcedure</c> being set to true, will create the stored procedure with a custom name
+            ?storedProcedureName: string,
             /// <c>true</c> to inhibit logging of client name
-            [<O; D null>]?skipLog) : CosmosClient =
+            [<O; D null>]?skipLog) : EquinoxCosmosClient =
+
         let (Discovery.UriAndKey (databaseUri=uri; key=key)) = discovery
         if skipLog <> Some true then logName uri name
-        new CosmosClient(string uri, key, __.ClientOptions)
+        let cosmosClient = new CosmosClient(string uri, key, __.ClientOptions)
 
-    /// Yields a Connection configured per the specified strategy
-    member __.Connect
-        (   /// Name should be sufficient to uniquely identify this connection within a single app instance's logs
-            name, discovery : Discovery,
-            /// <c>true</c> to inhibit OpenAsync call
-            [<O; D null>]?skipOpen,
-            /// <c>true</c> to inhibit logging of client name
-            [<O; D null>]?skipLog) : Async<Connection> = async {
-        let client = __.CreateClient(name, discovery, ?skipLog=skipLog)
-        // TODO validate this is equivalent to forcing a connect
-        if skipOpen <> Some true then let! _ = client.ReadAccountAsync() |> Async.AwaitTaskCorrect in ()
-        return Connection(client, ?readRetryPolicy=readRetryPolicy, ?writeRetryPolicy=writeRetryPolicy) }
+        match provisioningMode with
+        | Some mode ->
+            EquinoxCosmosInitialization.initializeContainer cosmosClient dbName containerName mode (defaultArg createStoredProcedure true) storedProcedureName None
+            |> Async.Ignore
+            |> Async.RunSynchronously
```

Comment:
What are your thoughts on it? I figured that it's ok to do the init here if the consumer provides args in the method call.

Reply:
The creating/updating of the container RUs and/or making sure the proc is up to date is something that's logically quite separated from the act of connecting

The key concern is that a CreateClient call winds up doing lots more that that

I believe it should be quite viable to separate that secondary function into an (Async) member somewhere else

Reply:
The issue is that in order to initialize the collection, you need to have a `CosmosClient`, and conversely, in order to create an `EquinoxCosmosClient`, you need to have an already-initialized container :) ... That's why I put it as part of this method. If you think it's a big concern, I could theoretically create a 2nd method on the factory called `InitializeContainerAndCreateClient`. Would that solve your concern?

Reply:
The initialization is deliberately lazy as the original impl provided for a single client to be used with a single context which can map a stream name to N containers
While I think that behavior and the complexity it brings is something that's probably a good thing to jettison, I'd still like to see the decoupling of creating a CosmosClient from the running of the stored proc

pushing out any asyncness as far as possible is definitely something I'd like to see

either way, regardless of whether we do it in here or not, I really want to avoid there being any Async.RunSynchronously calls in the impl (they are acceptable e.g. in `Equinox.Tool` as a pragmatic hack etc)

Reply:
Changed so that `EquinoxCosmosOperations` now has an `InitializeContainer()` method that can be explicitly called to create the container. Accordingly, its reference to `CosmosContainer` is now lazy and no longer provided in the constructor.

Reply:
🏩 :shipit: